### PR TITLE
Expanded update of security attributes on named pipe (fixes #54)

### LINKED ISSUE #54: Frida 6.0.8 cannot spawn or atach on Win7 x64
I have encountered an issue with Frida 6.0.8 on fully patched Win 7 SP1 x64 while trying to instrument any local process.
The spawn or attach command will fail with the same error:
```
C:\>frida -f %WINDIR%\system32\calc.exe
    _____
   (_____)
    |   |    Frida 6.0.8 - A world-class dynamic instrumentation framework
    |   |
    |`-'|    Commands:
    |   |        help      -> Displays the help system
    |   |        object?   -> Display information about 'object'
    |   |        exit/quit -> Exit
    |   |
    |   |    More info at http://www.frida.re/docs/home/
    `._.'

Failed to attach: error opening named pipe (ConvertStringSecurityDescriptorToSecurityDescriptor returned 0x00000539)
```
As per MSDN:
0x00000539 = ERROR_INVALID_SID [1337 (0x539) The security ID structure is invalid.]

### PR DESCRIPTION
**Pull request by riverar**
Windows XP-Windows 10 xplat ConvertStringSecurityDescriptorToSecurityDescriptor support, fixing
regression introduced (see #54)

### CODE REVIEW

File: lib/pipe/pipe-windows.c [hunk 1]
```diff
@@ -328,57 +332,14 @@ frida_pipe_open (const gchar * name, FridaPipeRole role, GError ** error)
...
-    sd = LocalAlloc (LPTR, SECURITY_DESCRIPTOR_MIN_LENGTH);
-    CHECK_WINAPI_RESULT (sd, !=, NULL, "LocalAlloc");
-
-    success = InitializeSecurityDescriptor (sd, SECURITY_DESCRIPTOR_REVISION);
-    CHECK_WINAPI_RESULT (success, !=, FALSE, "InitializeSecurityDescriptor");
-
-    success = SetSecurityDescriptorDacl (sd, TRUE, acl, FALSE);
-    CHECK_WINAPI_RESULT (success, !=, FALSE, "SetSecurityDescriptorDacl");
-  }
+  sddl = frida_pipe_get_sddl_string_for_os();
```
Comment (oleavr) [inline lib/pipe/pipe-windows.c]: Missing space before `(`.

File: lib/pipe/pipe-windows.c [hunk 2]
```diff
@@ -328,57 +332,14 @@ frida_pipe_open (const gchar * name, FridaPipeRole role, GError ** error)
...
+  success = ConvertStringSecurityDescriptorToSecurityDescriptor (sddl, SDDL_REVISION_1, &sd, NULL);
+  CHECK_WINAPI_RESULT (success, != , FALSE, "ConvertStringSecurityDescriptorToSecurityDescriptor");
```
Comment (oleavr) [inline lib/pipe/pipe-windows.c]: Left-over space after `!=`.

File: lib/pipe/pipe-windows.c [hunk 3]
```diff
@@ -464,3 +421,61 @@ frida_pipe_path_from_name (const gchar * name)
 
   return path;
 }
+
+static LPCWSTR
+frida_pipe_get_sddl_string_for_os (void)
+{
+    #define DACL_START_NOINHERIT L"D:PAI"
```
Comment (oleavr) [inline lib/pipe/pipe-windows.c]: Should stick to 2 space indentation.

File: lib/pipe/pipe-windows.c [hunk 4]
```diff
@@ -464,3 +421,61 @@ frida_pipe_path_from_name (const gchar * name)
...
+}
+
+static BOOL
+frida_pipe_is_windows_8_or_greater (void)
+{
+    return frida_pipe_is_windows_version_or_greater (6, 2, 0);
+}
+
+static BOOL
+frida_pipe_is_windows_version_or_greater (DWORD major, DWORD minor, DWORD servicePack)
```
Comment (oleavr) [inline lib/pipe/pipe-windows.c]: Variables should always be lower-case, i.e. `service_pack` here.

File: lib/pipe/pipe-windows.c [hunk 5]
```diff
@@ -464,3 +421,61 @@ frida_pipe_path_from_name (const gchar * name)
...
+{
+    OSVERSIONINFOEXW osvi;
+    ULONGLONG conditionMask;
```
Comment (oleavr) [inline lib/pipe/pipe-windows.c]: Variables should always be lower-case, i.e. `condition_mask` here.

**Comment by oleavr** [comment]
Just those nitpicks and this is ready to be merged. Awesome work!

**Comment by riverar** [comment]
Corrected.

File: lib/pipe/pipe-windows.c [hunk 6]
```diff
@@ -47,6 +46,11 @@ static WCHAR * frida_pipe_path_from_name (const gchar * name);
 static gboolean frida_pipe_backend_await (FridaPipeBackend * self, HANDLE complete, HANDLE cancel, GCancellable * cancellable, GError ** error);
 static void frida_pipe_backend_on_cancel (GCancellable * cancellable, gpointer user_data);
 
+static BOOL frida_pipe_is_windows_version_or_greater (DWORD major, DWORD minor, DWORD servicePack);
```
Comment (oleavr) [inline lib/pipe/pipe-windows.c]: One left-over camelCase argument here.

**Comment by riverar** [comment]
Squashed last camelCase variable that was left behind

**Comment by oleavr** [comment]
Thanks!

**Pull request merged**